src/coreComponents/physicsSolvers/contact/SolidMechanicsALMJumpUpdateKernels.hpp

```diff
@@ -0,0 +1,275 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SolidMechanicsALMUpdateKernels.hpp
+ */
+
+#ifndef GEOS_PHYSICSSOLVERS_CONTACT_SOLIDMECHANICSALMUPDATEKERNELS_HPP_
+#define GEOS_PHYSICSSOLVERS_CONTACT_SOLIDMECHANICSALMUPDATEKERNELS_HPP_
+
+#include "SolidMechanicsALMKernelsBase.hpp"
+
+namespace geos
+{
+
+namespace solidMechanicsALMKernels
+{
+
+/**
+ * @copydoc geos::finiteElement::ImplicitKernelBase
+ */
+template< typename CONSTITUTIVE_TYPE,
+          typename FE_TYPE >
+class ALMJumpUpdate :
+  public ALMKernelsBase< CONSTITUTIVE_TYPE,
```

Comment:
I may be wrong but I don't think this Kernel needs to know the constitutive_type. You can write it's own launch if you only need to dispatch on the fe_type.